src/confluent/tools/handlers/metrics/get-topic-metrics-handler.ts

```diff
@@ -0,0 +1,110 @@
+import { z } from "zod";
+import { MetricHandler } from "./metric-handler.js";
+import { ToolConfig } from "@src/confluent/tools/base-tools.js";
+import { ToolName } from "@src/confluent/tools/tool-name.js";
+import { ClientManager } from "@src/confluent/client-manager.js";
+import { CallToolResult } from "@src/confluent/schema.js";
+
+const getTopicMetricsArguments = z.object({
+  clusterId: z
+    .string()
+    .describe("The Kafka cluster ID (e.g., lkc-xxxxxx)")
+    .default(process.env.KAFKA_CLUSTER_ID || ""),
+  topicName: z.string().optional(), // Make topicName optional for filtering
+  metrics: z
+    .array(z.string())
+    .default([
+      "io.confluent.kafka.server/sent_records",
+      "io.confluent.kafka.server/received_records",
+      "io.confluent.kafka.server/sent_bytes",
+      "io.confluent.kafka.server/received_bytes",
+      "io.confluent.kafka.server/retained_bytes",
+      "io.confluent.kafka.server/consumer_lag_offsets",
+    ]),
+  intervalStart: z.string().optional(),
+  intervalEnd: z.string().optional(),
+  limit: z.number().optional(),
```

Comment:
We should default to a lower limit here, as some clients may have issues with context length. It would be a better experience to reduce it to say 10 instead of relying on the APIs default (100) so we don't accidently bloat contexts.

Example:
<img width="618" alt="image" src="https://github.com/user-attachments/assets/2628b177-538e-4ec9-92a4-9e0391ceae94" />